### ISSUE TITLE
array-bracket-spacing never not firing on object.property[ index ];

### ISSUE THREAD

**Issue by vvo** [eslint/eslint#2941]
When using `"array-bracket-spacing": [2, "never"]`, writing `object.property[ 1 ]` is allowed. I would expect it to fail like trying to `array[ 2 ]`

seems like a bug?

**Comment by gyandeeps** [comment]
* What version of eslint are you using?
* what kind of parse are you using?
* Can you share the code and the actual output?

I ran this code and I got the errors (using [online demo](http://eslint.org/demo/))
```js
/* eslint "array-bracket-spacing": [2, "never"] */
var foo = object.property[ 1 ];
```

report:

```sh
2:10 - "object" is not defined.
2:25 - There should be no space after '['
2:29 - There should be no space before ']'
```

**Comment by vvo** [comment]
Try this in online demo:

```js
/* eslint "array-bracket-spacing": [2, "never"] */
var obj = {hello: {mom: [1, 2]}};

obj.hello.mom[ 2 ] = "3";
```

**Comment by vvo** [comment]
Even this fails to report the error:

```js
/* eslint "array-bracket-spacing": [2, "never"] */
var obj = {hello: {mom: [1, 2]}};

obj.hello.mom[ 2 ] = "3";

obj.hello[ 0 ] = "ueaj";
```

**Comment by vvo** [comment]
Wonder if the local eslint config is taken into account, but reading the docs, array-bracket-spacing should be never by default

**Comment by gyandeeps** [comment]
I can still see the errors come on online demo for your above example
```js
/* eslint "array-bracket-spacing": [2, "never"] */
var obj = {hello: {mom: [1, 2]}};

obj.hello.mom[ 2 ] = "3";

obj.hello[ 0 ] = "ueaj";
```

```sh
1:50 - Expected linebreaks to be 'LF' but found 'CRLF'.
4:13 - There should be no space after '['
4:17 - There should be no space before ']'
6:9 - There should be no space after '['
6:13 - There should be no space before ']'
```

**Comment by vvo** [comment]
Are you using the livedemo or creating a script and running eslint on it? I do not have any error using the livedemo, duh?

**Comment by vvo** [comment]
![2015-07-07-160707_1228x317_scrot](https://cloud.githubusercontent.com/assets/123822/8547921/3da9b5d6-24c2-11e5-96be-4ea8eab9bbb2.png)

**Comment by gyandeeps** [comment]
I am using the live demo. Can you refresh your page and start again maybe.

**Comment by vvo** [comment]
Same, no issue whatsoever. Amazing. Could it be the linefeeds or smg messing up here?

**Comment by vvo** [comment]
```sh
> cat weird.js 
/* eslint "array-bracket-spacing": [2, "never"] */
var obj = {hello: {mom: [1, 2]}};

obj.hello.mom[ 2 ] = "3";

obj.hello[ 0 ] = "ueaj";

obj[ 2 ] = "weee";

var hello = [ 2 ];

> eslint -v
v0.24.0

> eslint weird.js

weird.js
  10:12  error  There should be no space after '['                                     array-bracket-spacing
  10:16  error  There should be no space before ']'                                    array-bracket-spacing
  11:0   error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  11:8   error  Missing semicolon                                                      semi

✖ 4 problems (4 errors, 0 warnings)

> node -v
v0.12.5
```

**Comment by vvo** [comment]
only one error shown when using `var`

**Comment by vvo** [comment]
property access is not warned

**Comment by gyandeeps** [comment]
We have unit test: https://github.com/eslint/eslint/blob/master/tests/lib/rules/array-bracket-spacing.js#L84-L92

**Comment by vvo** [comment]
How come I cannot make it error in the online demo nor in nodejs

**Comment by vvo** [comment]
Using your example:

![2015-07-07-162943_1296x320_scrot](https://cloud.githubusercontent.com/assets/123822/8548516/6dccb206-24c5-11e5-9685-a0c772c4f9fd.png)

**Comment by btmills** [comment]
@gyandeeps I'm seeing the same thing @vvo is seeing - `array-bracket-spacing` [doesn't check computed member expressions](https://github.com/eslint/eslint/blob/master/lib/rules/array-bracket-spacing.js#L155-L156). The tests you linked to only test that computed member expressions without spacing are valid, but they would still pass linting if there were space between those brackets. We don't have invalid tests for member expressions there at all.

Since we're deprecating `space-in-brackets`, I'm inclined to say this is an omission - though from what I'm not sure. Do we need a third rule to cover computed member expressions, or should they be part of `array-bracket-spacing` (despite the array name)? @xjamundx thoughts?

**Comment by gyandeeps** [comment]
hmmm... why is it giving errors on my machine on online demo. I am using windows 7 and chrome 43 if that makes a difference.

**Comment by btmills** [comment]
@gyandeeps perhaps you have demo rules configured differently from the defaults? Try an incognito tab.

**Comment by xjamundx** [comment]
https://github.com/eslint/eslint/blob/master/docs/rules/computed-property-spacing.md is probably what you want

**Comment by gyandeeps** [comment]
I got it, `space-in-brackets` was on for me so these errors are coming from that rule. Sorry about that.

**Comment by btmills** [comment]
@xjamundx thank you! I'm thinking this would be good to point out in the docs, and a PR to close this would:

- [ ] Add `computed-property-spacing` to the deprecation notice and related rules for `space-in-brackets`.
- [ ] Add `computed-property-spacing` to the related rules for `array-bracket-spacing`.
- [ ] Add `array-bracket-spacing` to the related rules for `computed-property-spacing`.

Does that sound good?

**Comment by xjamundx** [comment]
Yeah I'll throw something together by this evening.

**Comment by nzakas** [comment]
Just to clarify for everyone: `array-bracket-spacing` works only on array literals, so the reported behavior is correct. What you're looking for here is `computed-property-spacing`.

**Comment by vvo** [comment]
thanks all
